### PR TITLE
Ensure polygons are inside -180/180 range before making stac request

### PR DESCRIPTION
**Pull request by danielfdsilva**
See #732

I created a script to cut the polygons that cross the antimeridian and them shift them inside the correct range.
Includes some unit tests for the function.

### CODE REVIEW

**Comment by j08lue** [comment]
When I draw on the left-most or right-most out of four cycles of the Earth across the antimeridian, part of the geometry seems to be there but not visible:

Drawn on left-most cycle:

image

Drawn on right-most cycle:

image

**Comment by j08lue** [comment]
We earlier discussed whether we could just show only one cycle of the Earth. My concern then was, that this would make the map look weird. However, if this causes too much headache, could we perhaps show the whole map (with repetitions), but guide the user only draw within those cycles where we can do the normalization safely?

**Comment by danielfdsilva** [comment]
3 maps on a very wide and narrow window.
image

Full screen on my high-res monitor, with a smaller timeline allows to view the whole map
image

**Review by j08lue — APPROVED** [review]
Awesome, great work handling all these over-the-edge cases... The guardrails keep me from running into the original issue now.

File: app/scripts/utils/antimeridian.ts [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    // So we need to find out by how many times we need to shift 360. This will
+    // be defined by the smallest lng value. Any coordinates that are still
+    // outside after the shift will be shifted again in a later run.
+    const largestLng = Math.max.apply(
+      null,
+      ring.map(([lng]) => lng)
+    );
+
+    // Or 1 in case the largestLng is exactly -180.
+    const shift = Math.ceil(Math.abs((largestLng + 180) / 360)) || 1;
```
Comment (hanbyul-here) [inline app/scripts/utils/antimeridian.ts:147]: Is it okay to round up like this? 1 degree in Mercator doesn't seem to be that small number?

File: app/scripts/utils/antimeridian.test.ts [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+    };
+
+    const expected: Feature<MultiPolygon> = {
+      type: 'Feature',
+      properties: {},
+      geometry: {
+        coordinates: [
+          [
+            [
+              [-170, -20],
```
Comment (hanbyul-here) [inline app/scripts/utils/antimeridian.test.ts:30]: Can you point me where the order of the coordinates changes? I can't wrap my head around it.

**Review by hanbyul-here — APPROVED** [review]
Yay test. (Test description can be more helpful! ) I left a concern about an edge case and a question about the order of the coordinates. 🤔 I wonder if there are better names than `east` and `west` for coordinates to go over >180, <-180 . Nothing seems to be a blocker

File: app/scripts/utils/antimeridian.ts (hunk 1 above)
Comment (danielfdsilva) [reply app/scripts/utils/antimeridian.ts:147]: This does not refer to 1º but to how many times to ship 360º. The result is that the coordinates move to the "repeated map" they are to the correct one (between -180/180)

File: app/scripts/utils/antimeridian.test.ts (hunk 2 above)
Comment (danielfdsilva) [reply app/scripts/utils/antimeridian.test.ts:30]: That was also strange to me, but for some reason the turf.js function (intersect and difference), reorder the coordinates when cutting the polygons.

**Comment by danielfdsilva** [comment]
@hanbyul-here You're right. The messages were absolutely atrocious 😅. I made them more descriptive and will merge. Feel free to suggest further improvements.
I could not thing of better names than west/east, but happy to hear a suggestion.
